articles/active-directory/develop/quickstarts/active-directory-ios.md

```diff
@@ -0,0 +1,180 @@
+
+```bash
+echo "github \"AzureAD/microsoft-authentication-library-for-objc\" \"master\"" > Cartfile
+carthage update
+```
+
+Then, in XCode, open the **General**, go to **Linked Frameworks and Libraries** and add **MSAL.framework** in **Add other...**, then add the following content in **New Run Script Phase**:
+
+```text
+/usr/local/bin/carthage copy-frameworks
+```
+
+And the following to **Input Files**:
+
+```text
+$(SRCROOT)/Carthage/Build/iOS/MSAL.framework
+```
+
+### MSAL initialization
+
+You can add the reference for MSAL by adding the following code:
+
+```swift
+import MSAL
+```
+
+Then, initialize MSAL using the following code:
+
+```swift
+self.applicationContext = try MSALPublicClientApplication.init(clientId: kClientID, authority: kAuthority)
+```
+
+> |Where: ||
+> |---------|---------|
+> | `clientId` | The Application ID from the application registered in *portal.microsoft.com* |
+> | `authority` | The Azure AD v2.0 endpoint. In most of cases this will be *https<span/>://login.microsoftonline.com/common/v2.0* |
+
+### Requesting tokens
+
+MSAL has two methods used acquire tokens: `acquireToken` and `acquireTokenSilent`.
+
+#### Getting a user token interactively
+
+Some situations require forcing users to interact with Azure Active Directory (Azure AD) v2.0 endpoint which will result in a context switch to the system browser to either validate users's credentials or for consent. Some examples include:
+
+* The first time users sign in to the application
+* When users may need to reenter their credentials because the password has expired
+* When your application is requesting access to a resource that the user needs to consent to
+* When two factor authentication is required
+
+```swift
+self.applicationContext.acquireToken(forScopes: self.kScopes)
+```
+
+> |Where:||
+> |---------|---------|
+> | `forScopes` | Contains the scopes being requested (that is, `{ "user.read" }` for Microsoft Graph or `{ "api://<Application ID>/access_as_user" }` for custom Web APIs) |
+
+#### Getting a user token silently
+
+You don't want to require the user to validate their credentials every time they need to access a resource. Most of the time you want token acquisitions and renewal without any user interaction. You can use the `AcquireTokenSilentAsync`method to obtain tokens to access protected resources after the initial `AcquireTokenAsync` method:
+
+```swift
+self.applicationContext.acquireTokenSilent(forScopes: self.kScopes, user: applicationContext.users().first)
+```
+
+> |Where: ||
+> |---------|---------|
+> | `forScopes` | Contains the scopes being requested (that is, `{ "user.read" }` for Microsoft Graph or `{ "api://<Application ID>/access_as_user" }` for custom Web APIs) |
```

Comment:
Since input scopes is an array of strings, let's use an array in this sample.

So:

`[ "user.read" ]` and `[ "api://<Application ID>/access_as_user" ]`

Reply:
@oldalton - not sure why GitHub removed your comment - what do you think about the lines for *acquireTokenSilent* and *acquireToken* ?